enh-ruby-mode.el

```diff
@@ -891,6 +891,14 @@ not treated as modifications to the buffer."
            (1- (length (match-string-no-properties 0))))
       0))
 
+(defun enh-ruby-continue-p (prop)
+  "Return whether PROP is a continue property."
+  (eq 'c prop))
+
+(defun enh-ruby-point-continue-p (point)
+  "Return whether property at POINT is a continue property."
+  (enh-ruby-continue-p (get-text-property point 'indent)))
+
```

Comment:
There's _probably_ some macro thing that can be done to clean create more in a clean declarative fashion...but this is a great start. Thank you.

Reply:
@zenspider Maybe something like this:

```lisp
(defmacro enh-ruby-create-indent-query-defuns (legible-prop real-prop)
  "Create macros to query for REAL-PROP indent property through LEGIBLE-PROP."
  (let* ((legible-prop (symbol-name legible-prop))
         (real-prop real-prop)
         (prop-fun-name (concat "enh-ruby-" legible-prop "-p"))
         (point-fun-name (concat "enh-ruby-point-" legible-prop "-p")))
    `(progn
       (defmacro ,(intern prop-fun-name) (prop)
         (list 'eq '',real-prop prop))
       (defmacro ,(intern point-fun-name) (point)
         (list 'eq '',real-prop `(get-text-property ,point 'indent))))))

(enh-ruby-create-indent-query-defuns continue c)
(enh-ruby-create-indent-query-defuns open l)
(enh-ruby-create-indent-query-defuns close r)
(enh-ruby-create-indent-query-defuns stmt-begin b)
(enh-ruby-create-indent-query-defuns block-begin d)
(enh-ruby-create-indent-query-defuns block-end e)
(enh-ruby-create-indent-query-defuns backdent s)

;; Examples:

(enh-ruby-backdent-p prop)
(enh-ruby-end-p prop)
(enh-ruby-point-continue-p (point))
(enh-ruby-point-close-p (point))
(enh-ruby-point-open-p (save-excursion ... (point)))
;; Etc...
```

What do you think? The upside is that the code is injected through a macro and there's no runtime cost.